storage.go

```diff
@@ -349,6 +479,34 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
 	if opt.HasStorageClass {
 		input.StorageClass = &opt.StorageClass
 	}
+	if opt.HasExceptedBucketOwner {
+		input.ExpectedBucketOwner = &opt.ExceptedBucketOwner
+	}
+	if opt.HasServerSideEncryptionBucketKeyEnabled {
+		input.BucketKeyEnabled = &opt.ServerSideEncryptionBucketKeyEnabled
+	}
+	if opt.HasServerSideEncryptionCustomerAlgorithm {
+		input.SSECustomerAlgorithm = &opt.ServerSideEncryptionCustomerAlgorithm
+	}
+	if opt.HasServerSideEncryptionCustomerKey {
```

Comment:
Maybe we can introduce a function as we dose in https://github.com/aos-dev/go-service-qingstor/pull/38/files#diff-8b7f25bac2ae04c2ce2abb628c6aa28103fd113103a62f3e9d232fa995684915R393-R402

```go
calculateEncryptionHeaders(algo string, key []byte) (algorithm, keyBase64, keyMD5Base64 *string, err error)
```

We should do the key length check in this function